### PR TITLE
[CPO] Check that `max_prompt_length < max_length`

### LINKED ISSUE #3339: Loss becomes NaN when response is empty due to `max_prompt_length >= max_length` in SimPO
Encountered, `NaN` while training using SimPO under condition max_prompt_length == max_length.

**Details**:
```
Algorithm : SimPO
max_prompt_length: 2048
max_length: 2048
```

**Issue**:
The normalization for loss is done using the number of non-pad tokens in the response. If the response is empty, this results in division by zero i.e., `0/0`, causing `loss = NaN`.

**Expected Behavior**:
- TRL should skip or filter out samples with an empty response to avoid `NaN` losses
**OR**
- The implementation should enforce `max_prompt_length < max_length` during data preparation or training.
**OR**
- Clamp denominator to 1.0 when response length is empty to avoid division by zero at [cpo_trainer.py](https://github.com/huggingface/trl/blob/2bc182c4fb8180a3cb815c61ca6c528a86559d14/trl/trainer/cpo_trainer.py#L717)

### PR DESCRIPTION
**Pull request by LeonEricsson**
Fixes #3339 

As mentioned in the issue I'm not sure this is necessary, the docs are pretty clear already but I guess it doesn't hurt?

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by qgallouedec** [comment]
It doesn't hurt

**Review by qgallouedec — APPROVED** [review]
